### PR TITLE
Adding public access filtering

### PR DESCRIPTION
**Pull request by Patrik-Netent**

### CODE REVIEW

File: modules/network-firewall/variables.tf [hunk 1]
```diff
@@ -28,3 +28,8 @@ variable "name_prefix" {
   type        = string
 }
 
+variable allowed_public_networks {
+  description = "The public networks that is allowed access to public_subnetwork"
+  default = ["0.0.0.0"]
```
Comment (veggiemonk) [inline modules/network-firewall/variables.tf]:
```suggestion
  default = ["0.0.0.0/0"]
```

File: modules/network-firewall/variables.tf [hunk 2]
```diff
@@ -28,3 +28,8 @@ variable "name_prefix" {
...
+  type = list(string)
+}
```
Comment (veggiemonk) [inline modules/network-firewall/variables.tf]:
```suggestion
}

```

File: modules/vpc-network/main.tf [hunk 3]
```diff
@@ -115,8 +115,8 @@ module "network_firewall" {
 
   project = var.project
   network = google_compute_network.vpc.self_link
+  allowed_public_networks = var.allowed_public_networks
 
   public_subnetwork  = google_compute_subnetwork.vpc_subnetwork_public.self_link
   private_subnetwork = google_compute_subnetwork.vpc_subnetwork_private.self_link
 }
-
```
Comment (veggiemonk) [inline modules/vpc-network/main.tf]: ?

File: modules/vpc-network/variables.tf [hunk 4]
```diff
@@ -65,3 +65,8 @@ variable "enable_flow_logging" {
   default     = true
 }
 
+variable allowed_public_networks {
+  description = "The public networks that is allowed access to public_subnetwork"
+  default = ["0.0.0.0"]
+  type = list(string)
+}
```
Comment (veggiemonk) [inline modules/vpc-network/variables.tf]:
```suggestion
}

```

File: modules/vpc-network/variables.tf [hunk 5]
```diff
@@ -65,3 +65,8 @@ variable "enable_flow_logging" {
   default     = true
 }
 
+variable allowed_public_networks {
+  description = "The public networks that is allowed access to public_subnetwork"
+  default = ["0.0.0.0"]
```
Comment (veggiemonk) [inline modules/vpc-network/variables.tf]:
```suggestion
  default = ["0.0.0.0/0"]
```

File: modules/network-firewall/variables.tf (hunk 1 above)
Comment (veggiemonk) [reply modules/network-firewall/variables.tf]: it is a CIDR

File: modules/network-firewall/variables.tf (hunk 2 above)
Comment (Patrik-Netent) [reply modules/network-firewall/variables.tf]: Good catch

File: modules/vpc-network/main.tf (hunk 3 above)
Comment (Patrik-Netent) [reply modules/vpc-network/main.tf]: ?

**Pull request merged**